pennylane/beta/tapes/qnode.py

```diff
@@ -395,8 +414,16 @@ def __call__(self, *args, **kwargs):
         # construct the tape
         self.construct(args, kwargs)
 
+        if self._caching:
+            self.qtape._cache_execute = self._cache_execute

```

Comment:
How come a copy of the private method is made here?

Reply:
I have added a comment explaining